clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -536,9 +536,34 @@ DiagnoseHLSLAvailability::FindAvailabilityAttr(const Decl *D) {
 void DiagnoseHLSLAvailability::CheckDeclAvailability(NamedDecl *D,
                                                      const AvailabilityAttr *AA,
                                                      SourceRange Range) {
-  if (ReportOnlyShaderStageIssues && !AA->getEnvironment())
-    return;
 
+  IdentifierInfo *IIEnv = AA->getEnvironment();
+
+  if (!IIEnv) {
+    // The availability attribute does not have environment -> it depends only
+    // on shader model version and not on specific the shader stage.
+
+    // Skip emitting the diagnostics if the diagnostic mode is not set to
+    // strict (-fhlsl-strict-availability) because all relevant diagnostics
+    // were already emitted in the DiagnoseUnguardedAvailability scan
+    // (SemaAvailability.cpp).
+    if (SemaRef.getLangOpts().HLSLStrictAvailability)
+      return;
```

Comment:
Does returning here mean that it skips emitting diagnostics?  If so, it looks like the code skips emitting diagnostics in strict mode, which is the opposite of what I think the comment says it is doing.

Reply:
Yes, it skips emitting diagnostic here in strict mode because it has already been emitted in the `DiagnoseUnguardedAvailability` phase during parsing. If it did not there would be duplicate messages.

Reply:
It's the opposite of the condition in SemaAvailability.cpp:164 which you commented on above.

Reply:
> Yes, it skips emitting diagnostic here in strict mode because it has already been emitted in the...

Right, but the comment says it skips emitting the diagnostics if it is _not_ in strict mode.

Reply:
Oh yes! Sorry :) I'll update the comment.